cosign/README.md

```diff
@@ -0,0 +1,137 @@
+# Cosign Verifier
+
+## Introduction
+
+In order to provide signatures over arbitrary artifacts, Sigstore performs three major operations:
+
+1. Introduce an OpenID Connect (OIDC) Provider, assure that a client is in control of an identity
+2. Issue short-lived public-key certificates with these identities (from a Certificate Authority), and publish these certificates to an Identity Log.
+3. Publish a signature over an artifact (or artifact meta-data) to an Artifact Log, allowing verifiers to check its validity.
+
+The first and second operations are performed by a system called **Fulcio**, which functions as a Certificate Authority and transparency log for a namespace of OIDC identities (Identity Log). The third operation is performed by a system called **Rekor**, a transparency log for artifact signatures (Artifact Log). And **Cosign** is the reference Sigstore client implementation to sign arbitrary artifacts. With these components, Sigstore completed trust setup, signing and verification.
+
+Sigstore compares a tuple of signature, key/certificate, and artifact from the timestamped object against the timestamped Rekor entry. If they match, it confirms that the signature is valid because the user knows that the expected software creator, whose identity was certified at the moment of signing, published the software artifact in their possession.
+
+```mermaid
```

Comment:
the graph lg overall, I'd like to add more details to it:
1. make it more clear so that we can find the entrypoint of the graph easily, at least it takes me a while to find the first step(Authenticates with).
2. Use consistent step sequence, using number or alphabet consistently.
3. I would take interest in the data that exchanges between different entities, like what's fetched/uploaded in repository.
4. We can add a note section to explain some essential steps since we cannot desribe everything in the graph.
5. This design is focusing on the cosign-verifer for ratify, we should add more details to the verification workflow. I'd like to know how we validate the signature, validate the cert and so on.
6. And I didn't see the artifact log in the graph.

Reply:
Thanks for the review and clear next steps suggestions. Working on an improvement.

Reply:
Updated.